File: 1408. String Matching in an Array/solution.js
/**
 * @param {string[]} words
 * @return {string[]}
 */
 var stringMatching = function(words) {
  words.sort(function(a, b){return a.length - b.length});
  let result = [];
  for (let index = 0; index < words.length; index++){
    for (let index2 = index + 1; index2 < words.length; index2++){
      if(words[index2].includes(words[index])){
        result.push(words[index]);
        break;
      }
    }
  }

  return result;
};

File: 1408. String Matching in an Array/solution.cpp
class Solution {
public:
  vector<string> stringMatching(vector<string>& words) {
    vector<string> result;
    for (int index = 0; index < words.size(); index++){
      for (int index2 = 0; index2 < words.size(); index2++){
        if (words[index] != words[index2] && words[index2].find(words[index]) != -1){
          result.push_back(words[index]);
          break;
        }
      }
    }

    return result;
  }
};